docs/contributing.md

```diff
@@ -0,0 +1,119 @@
+# Contributing to Wagtail Vector Index
+
+## Getting the Code
+
+To make changes to this project, first clone this repository:
+
+```shell
+git clone https://github.com/wagtail/wagtail-vector-index.git
+cd wagtail-vector-index
+```
+
+## Setting up your development environment
+
+The easiest way to bootstrap your development environment is using `tox`. We recommending installing this in an isolated environment using `pipx`:
+
+```shell
+python -m pip install pipx-in-pipx --user
+pipx install tox
+```
+
+Alternative installation methods can be found in the [tox documentation](https://tox.wiki/en/latest/installation.html).
+
+Once installed, create a development virtual environment with:
+
+```shell
+tox devenv -e interactive
+. ./venv/bin/activate
+```
+
+### Using devcontainers
+
+Alernatively, a [devcontainer](https://containers.dev/) configuration is available in this repository with `tox` configured.
+
+Using this devcontainer in VSCode will automatically enable the virtual environment, in other devcontainer environments you will need to activate it with `. ./venv/bin/activate`.
+
+## Working with the test application
+
+A Wagtail example for testing/development is bundled in this repo (find it at `tests/testapp`).
+
+You can interact with this application inside your virtual environment using the `testmanage.py` script as you would a normal Django/Wagtail app.
+
+For example, to bring up a development server on port `8000`, run:
+
+```shell
+python testmanage.py runserver 0:8000
+```
+
+If you have bootstrapped your environment with `tox`, there will be a default admin user (username: `admin`, password: `changeme`) available.
+
+### Developing with the pgvector backend
+
+By default, the development environment runs using the `NumpyBackend` but if you need to test with large datasets or work on the `PgvectorBackend` itself, you'll need a local PostgreSQL instance with `pgvector` installed.
+
+This project comes bundled with a `docker-compose.yml` file which can bring up a suitable database instance for you.
+
+To use it:
+
+1. Install [Docker](https://docs.docker.com/engine/install/)
+2. Run `docker-compose up -d`
+
+You can then configure the test application to use this database by setting the following environment variables.
+
+```
+WAGTAIL_VECTOR_INDEX_DEFAULT_BACKEND=pgvector
+DATABASE_URL=postgres://postgres:postgres@localhost:5432/postgres
+```
+
+If you're using the devcontainer, this database service is automatically started for you.
+
+## Working with your own application
+
+If you already have an application you'd like to use when developing Wagtail Vector Index, you can install the package directly from source alongside it using `flit`:
+
+```
+# Install flit
+python -m pip install flit
+# Change directory to where you have cloned the wagtail-ai repo
+cd wagtail-vector-index
+# Install the package using 'symlink' mode so you can change the code without having to reinstall it
+flit install -s
```

Comment:
For whatever reason on my local computer (MacOS) invoking `flit` inside a venv does not install stuff into the venv. I have to use `python -m venv` which is quite weird. I wonder if this is just specific to something wrong on my computer or affects everyone?

@zerolab did you have the same issue?

Reply:
Is it possible you have flit installed globally? if so, that would explain the behaviour.

I believe you need to invoke it with `python -m flit` (for it is `py` which is aliased to python3) so that it picks up the venv